internal/http/server/server_handler.go

```diff
@@ -109,36 +105,129 @@ func (h *Handler) UpdateMetric(w http.ResponseWriter, r *http.Request) {
 }
 
 func (h *Handler) MainPage(w http.ResponseWriter, r *http.Request) {
-	if r.Method == http.MethodGet {
-		urlPart := strings.Split(r.URL.Path, "/")
-		if len(urlPart) > 2 {
-			http.Error(w, "Wrong URL", http.StatusBadRequest)
-			return
-		}
-		if len(urlPart[1]) != 0 {
-			http.Error(w, "Wrong URL", http.StatusBadRequest)
-			return
-		}
-		wd, err := os.Getwd()
-		if err != nil {
-			panic(err)
-		}
-		tmplFile := filepath.Join(filepath.Dir(wd), h.config.PathToHTMLTemplate)
-		tmpl, err := template.New(tmplFile).ParseFiles(tmplFile)
+	w.Header().Set("Content-Type", "text/html; charset=utf-8")
+	urlPart := strings.Split(r.URL.Path, "/")
+	if len(urlPart) > 2 {
+		http.Error(w, "Wrong URL", http.StatusBadRequest)
+		return
+	}
+	if len(urlPart[1]) != 0 {
+		http.Error(w, "Wrong URL", http.StatusBadRequest)
+		return
+	}
+	wd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	tmplFile := filepath.Join(filepath.Dir(wd), h.config.PathToHTMLTemplate)
```

Comment:
CHANGE REQUEST: it won't work if you move it into a Docker container. You need to embed this file.

Reply:
done